src/Type/IntersectionType.php

```diff
@@ -206,12 +206,32 @@ private function describeItself(VerbosityLevel $level, bool $skipAccessoryTypes)
 	{
 		$typesToDescribe = [];
 		$skipTypeNames = [];
+
+		$nonEmptyStr = false;
+		$nonFalsyStr = false;
 		foreach ($this->getSortedTypes() as $type) {
 			if ($type instanceof AccessoryNonEmptyStringType
 				|| $type instanceof AccessoryLiteralStringType
 				|| $type instanceof AccessoryNumericStringType
 				|| $type instanceof AccessoryNonFalsyStringType
 			) {
+				if ($type instanceof AccessoryNonFalsyStringType) {
```

Comment:
This doesn't need hacks in IntersectionType.

* AccessoryNonEmptyStringType = AccessoryNonFalsyStringType + '0'
* AccessoryNonFalsyStringType = AccessoryNonEmptyStringType - '0'

This means that AccessoryNonFalsyStringType is "smaller".

* `AccessoryNonFalsyStringType->isSuperTypeOf(AccessoryNonEmptyStringType)` = maybe (makes sense, not every AccessoryNonEmptyStringType is AccessoryNonFalsyStringType)
* `AccessoryNonEmptyStringType->isSuperTypeOf(AccessoryNonFalsyStringType) = yes` (makes sense, all AccessoryNonFalsyStringType are AccessoryNonEmptyStringType)

Reply:
`isSuperTypeOf` implemented in [45c5c6a](https://github.com/phpstan/phpstan-src/pull/1542/commits/45c5c6af97d490b9970da704a7d0345e3b1bddf3).

the hacks we see here are required to make sure a intersection cannot contain `non-falsy-string` and `non-empty-string` at the same time.

I guess this need to be handled via `isSubTypeOf` or `tryRemove` or similar instead, to get rid of these `IntersectionType` changes

Reply:
I think its finally covered with [30728d4](https://github.com/phpstan/phpstan-src/pull/1542/commits/30728d413fa29d81416bb5e49b22670244cb3635) and [7fd0884](https://github.com/phpstan/phpstan-src/pull/1542/commits/7fd08842406c896fa2b875fb8c791e9e3005bc23)

Reply:
I cannot get rid of this hack in `IntersectionType` atm, without regressing tests.